packages/vhd-lib/src/Vhd/VhdAbstract.integ.spec.js

```diff
@@ -0,0 +1,140 @@
+/* eslint-env jest */
+
+import rimraf from 'rimraf'
+import tmp from 'tmp'
+import fs from 'fs-extra'
+import { getSyncedHandler } from '@xen-orchestra/fs'
+import { Disposable, pFromCallback } from 'promise-toolbox'
+
+import { openVhd } from '../index'
+import { createRandomFile, convertFromRawToVhd, createRandomVhdDirectory } from '../tests/utils'
+import { VhdAbstract } from './VhdAbstract'
+
+let tempDir
+
+jest.setTimeout(60000)
+
+beforeEach(async () => {
+  tempDir = await pFromCallback(cb => tmp.dir(cb))
+})
+
+afterEach(async () => {
+  await pFromCallback(cb => rimraf(tempDir, cb))
+})
+
+test('It creates an alias', async () => {
+  await Disposable.use(async function* () {
+    const handler = yield getSyncedHandler({ url: 'file:///tmp/' })
+    const tempDirFromRoot = tempDir.slice(5) // remove /tmp/
+    const aliasPath = `${tempDirFromRoot}/alias.alias.vhd`
+
+    const testOneCombination = async ({ targetPath, targetContent }) => {
+      await VhdAbstract.createAlias(handler, aliasPath, targetPath)
+      // alias file is created
+      expect(await fs.exists('/tmp/' + aliasPath)).toEqual(true)
+      // content is the target path relative to the alias location
+      const content = await fs.readFile('/tmp/' + aliasPath, 'utf-8')
```

Comment:
Why not using `resolveAlias` instead?

This would test the API and keep the implementation self-contained.

Reply:
resolveAlias already have its own tests suite and I prefer to test the bare minimum code surface at once, especially since I'm working on both part simultaneously